### DISCUSSION TITLE
Features für den Alltag/Rollout

### DISCUSSION THREAD

**Discussion by fb4thxUall** [jzakotnik/openlibry#137] [Ideas]
Hallo zusammen.

Ich habe die OpenLibry-Installation heute an das Team der Schulbücherei übergeben mitsamt einer kleinen Einführung. Damit startet jetzt die produktive Betrieb. Zunächst muss der Buchbestand digital aufgenommen werden (das ist erst teilweise passiert), anschließend müssen alle Bücher mit Etiketten versehen werden.

Der erste Eindruck von OpenLibry war sehr gut, dennoch hatte sas Team sofort zwei Schwierigkeiten, aus denen diese Vorschläge hier resultieren:

1) Es sollte auf den Etiketten das Genre sichtbar sein. Denn: Wenn man ein Buch mit Etikett in der Hand hat, weiß man nicht, in welcher Abteilung man es einsortieren muss.

2) Es ist nur möglich, alle oder einen anfangenden Teil der Etiketten zu drucken. Das wäre aber sehr hilfreich, um die Arbeiten schrittweise erledigen zu können. Beispiel: "Diesen Monat etikettieren wir die Abteilung Sachbücher." 
Druckt man stattdessen jetzt die Etiketten komplett aus, muss man aus der gesamten Bibliothek zunächst mal das passende Buch finden, bekleben, repeat. Das ist sehr ineffizient. Unsere Bibliothek umfasst ca. 5000 Bücher. Es wäre also sehr praktisch, wenn man Etiketten für ein ausgewähltes Genre drucken könnte.

-> 3) Es resultiert daraus ein Folgeproblem: Die Genres sind aktuell nicht existent. Ich habe mir aktuell damit beholfen, dass ich "Schlagwörter" mit dem Genre befüllt habe. Es wäre mMn von Vorteil, dieses Element noch mit aufzunehmen, auch wenn das einige zusätzliche Änderungen in der DB und im Interface mit sich bringt.

Wie sehen das die Anderen/Jure?

LG

**Comment by fb4thxUall** [comment]
Ich ergänze mal selbst noch direkt was: 

Im laufenden Betrieb kommt es ja vor, dass man Bücher dazu bekommt oder einfach einzelne Etiketten nachdrucken muss. Deswegen sollte man darüber hinaus auch die Möglichkeit haben, zu einzelnen Büchern Etiketten zu drucken. Vielleicht mit einer Mehrfachauswahl.

**Comment by jzakotnik** [comment]
Ich finde die Vorschläge gut, Genre fehlt irgendwie, ich vermute das ist dadurch entstanden, dass es in OpenBiblio nicht entsprechend genutzt wurde.
Ich mache ein neues Issue auf und da können wir tracken was es alles braucht?

Nachtrag: erinnere mich wieder was mein mentales Problem mit „Genre“ war - es ist nur eins, während Schlagwörter verschiedene Perspektiven auf das gleiche Buch ermöglichen. Also ein Kinderbuch über Pferde und Freundschaft, ist das ein Kinderbuch, ein Pferdebuch oder ein Beziehungsbuch? Tags ermöglichen eine variable Zuordnung zu der Kategorie auf die man filtert. Ich glaube Genre stammt noch aus der physischen Zeit mit Kärtchen und so, wo ein Buch nur einer Kategorie zugeordnet werden konnte. In der digitalen Welt brauchts das eigentlich nicht.

Ich sehe aber ein dass die Bücherei-Welt so arbeitet insofern lass uns das pragmatisch in Angriff nehmen.

**Reply by jzakotnik** [reply]
Ein Workaround könnte sein, dass das 1. Schlagwort immer das Genre ist (also das „wichtigste Schlagwort“), was meint ihr?

**Comment by fb4thxUall** [comment]
So habe ich die Datenbank auch befüllt, um die Genre Info erstmal nicht zu verlieren. 
Als kurzfristige Lösung, um die Info mit auf dem Etikett drucken zu können, finde ich das gut. 

Langfristig ist es glaube ich aber klüger, das sauber  zu implementieren. So habe ich dich aber auch verstanden.

Man merkt an meiner Response Zeit sofort: Schule ist wieder voll da :-D



Am 23. August 2024 09:28:31 MESZ schrieb Jure Zakotnik ***@***.***>:
>Ein Workaround könnte sein, dass das 1. Schlagwort immer das Genre ist (also das „wichtigste Schlagwort“), was meint ihr?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/jzakotnik/openlibry/discussions/137#discussioncomment-10427075
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Reply by fb4thxUall** [reply]
Hab gerade deinen Post erst richtig ganz gelesen, merke ich. Ich habe da nochmal etwas drüber nachgedacht:

Nimmt man das erste Schlagwort als Genre, hat man die Funktionalität zunächst mal gegeben. Auch das Drucken nach "Genre/1. Schlagwort" ließe sich realisieren. Eventuell würde das sogar die Nutzbarkeit flexibler machen, wenn man Drucken nach Schlagwort macht. 
Dem entgegen steht, dass es langfristig Side-Effekts haben könnte. Aber man könnte es auch dann noch aufsplitten. Also insofern kann ich beiden Lösungen etwas abgewinnen. 

Das Drucken einzelner Etiketten könnte ja im Prinzip eine Kopie der Bücherübersicht sein, die aber auch Mehrfachauswahl anbietet. Fancy, aber extrem optional, ist natürlich, wenn man in der Druckmaske dann angibt, welche Etikettenpositionen man bedrucken will. Das ermöglicht einem, auch angefangene Blätter mit Etiketten weiter zu bedrucken. Gehört aber auch in die Kategorie: Kostet viel Zeit, bringt nicht so sehr viel. Würde ich eher nach hinten priorisieren.


Ich fange im Moment mal an, mich ein bisschen in TypeSkript einzuarbeiten, obwohl eigentlich überhaupt keine Kapazitäten da sind, aber irgendwie wäre es ja hilfreich, selbst produktiver beisteuern zu können.

**Reply by jzakotnik** [reply]
Hast schon Recht, Genre im Datenmodell könnte auf Dauer Sinn machen, die Ideen schließen sich ja nicht aus.

Der einfachste bzw. verständlichste Fix für die Etiketten könnte sein, direkt anzugeben nach welchem Kriterium diese sortiert werden. Beim PDF Druck kann man dann einfach einstellen, wieviele bzw. welche Seiten man drucken möchte. Das einzige was blöd wäre, man verschwendet ggf. paar Aufkleber von der Seite.. was meinst Du?

**Reply by fb4thxUall** [reply]
Ah, das ist natürlich auch eine gute Idee. Das dürfte meiner Einschtzung nach den Implementationsaufwand zunächst deutlich reduzieren und die Funktionalität trotzdem gut abbilden. 
Für das Handling wäre natürlich recht wichtig, dass man im PDF erkennt, wo der gewünschte Abschnitt startet. Sonst muss man manuell im PDF suchen, was trotz Suchfunktion ziemlich unpraktisch wäre.
Deswegen überlege ich, ob man nicht vielleicht doch direkt eine Eingrenzung implementiert, die so aussieht:

Man kann im Report-Bereich ein Stichwort eingeben und im PDF werden nur alle Etiketten mit Büchern erstellt, die dieses Stichwort beinhalten. Da müsste man intern "nur" die Stichwort-Suche zwischenschalten, die man wiederum auch noch von der schon vorhandenen Stichwortsuche recyclen könnte.

Sind meine Gedanken verständlich? Reden wir aneinander vorbei? :-D

**Reply by jzakotnik** [reply]
Voll gute Idee, das ist eigentlich am einfachsten zu lösen (so 1-Zeiler 😸 ). Sollen alle Stichwörter in dem Filter landen oder nur das erste? Oder Sortiert nach Position in der Stichwortliste, dann kann man die letzten Seiten bei Bedarf weg lassen.

**Reply by fb4thxUall** [reply]
Hm gute Frage. Wenn man alle Stichwörter reinnimmt, ist es universeller und dürfte eigentlich keine Einschränkung sein. Zumindest bei uns wäre es das nicht, wir haben eh nur ein Stichwort :-D

Ich überlege, ob es Sinn macht, nicht nur nach Stichwort zu filtern, sondern nach allem. Dann könnte man auch alle Label zu einem Autoren oder alle Label von einem Titel (von dem es mehrere Exemplare geben kann) drucken. Hätte auch den Vorteil, dass man damit automatisch ein einzelnes Etikett für nur ein Buch drucken kann, weil man einfach die BuchID eingeben könnte. Das spart die Implementation an anderer Stelle.

Oder man baut 2 Eingabefelder ein und realisiert beides, einmal nur nach Stichwort/Genre, einmal nach Komplettsuche :-D
